app/scripts/metamask-controller.js

```diff
@@ -1082,11 +1093,31 @@ export default class MetamaskController extends EventEmitter {
         SnapController: this.snapController,
         NotificationController: this.notificationController,
         ///: END:ONLY_INCLUDE_IN
+        ...resetOnRestartStore,
       },
       controllerMessenger: this.controllerMessenger,
     });
     this.memStore.subscribe(this.sendUpdate.bind(this));
 
+    // if this is the first time, clear the state of by calling these methods
+    const resetMethods = [
+      this.accountTracker.resetState,
+      this.txController.resetState,
+      this.messageManager.resetState,
+      this.personalMessageManager.resetState,
+      this.decryptMessageManager.resetState,
+      this.encryptionPublicKeyManager.resetState,
+      this.typedMessageManager.resetState,
+      this.swapsController.resetState,
+      this.ensController.resetState,
+      this.approvalController.clear.bind(this.approvalController),
+      // WE SHOULD ADD TokenListController.resetState here too. But it's not implemented yet.
+    ];
+
+    if (globalThis.isFirstTimeProfileLoaded === true) {
+      this.resetStates(resetMethods);
+    }
+
```

Comment:
@segun : should we restrict the above change to MV3 only ?

Reply:
We can't move the code from app-init, because that's the only file where `chrome.runtime.onStartup.addListener(() => {` makes sense. It's the only file that's loaded when chrome starts